addons/web/static/src/views/fields/properties/property_definition_selection.js

```diff
@@ -32,6 +33,36 @@ export class PropertyDefinitionSelection extends Component {
                 inputs[this.state.newOptionIndex].focus();
             }
         });
+
+        useSortable({
+            enable: () => {
+                if (!this.props.canChangeDefinition || this.props.readonly) {
+                    return false;
+                }
+
+                const element = document.activeElement;
+                if (
+                    element &&
+                    element.tagName === "INPUT" &&
+                    element.parentNode.classList.contains("o_field_property_selection_option") &&
+                    (!element.value || !element.value.length)
+                ) {
+                    // an empty option has the focus, and so it will be removed when
+                    // we will start moving, so we block the drag and drop in that case
+                    return false;
```

Comment:
I find it a bit sad and not very user friendly to now allow to move a newly created option.

Knowing that you can focus some option and press enter to place it where you want is a bit niche and "power user".
I think most users will naturally press "Add option", then type their text and move it.
But there is a good chance that they also move their option before typing (some people really like their chaussettes sorted :disguised_face: ), in which case it's frustrating as the option disappears.

I understand that this is due to the "blur" mechanism, which is triggered upon clicking on the sorting control, not much you can do.
However, we could change the blur into a global DOM click event, that checks that you clicked outside of the input AND not on the drag&drop element before removing the option.

A bit annoying to do, but if not too complicated I think it would be nice, maybe to discuss with Luc?

If unclear, don't hesitate to ping me :slightly_smiling_face:

Reply:
> I find it a bit sad and not very user friendly to now allow to move a newly created option.

You can move a new option, if it's not empty (you need to edit it first)

See: https://drive.google.com/file/d/1wpP-Zeuv7FQsBfwsRpIAJ7_y29ewiVQ5/view?usp=sharing

> I think most users will naturally press "Add option", then type their text and move it.

So it will work, as in the video

> However, we could change the blur into a global DOM click event, that checks that you clicked outside of the input AND not on the drag&drop element before removing the option.

But if you start drag & drop an empty option, it won't be removed, but then you cancel the drag & drop (right click e.g.), are you are in a weird state with an empty option, so we also need to add a global event listener for "mouse up", etc 


So the only use case is: you click "Add an option" and you want to move the empty input, but not sure it's worth it to support that, considering the technical cost

Reply:
Yes, I understand that it works when you type before moving, sorry if it wasn't clear in my message.

I just wanted to see if we could handle the case: I add and I try moving it to the right spot before typing.
But if you think it's niche and not worth the cost, I'm ok with that :slightly_smiling_face: